src/Core/Resolvers/Factories/QueryManagerFactory.cs

```diff
@@ -102,6 +102,11 @@ private void ConfigureQueryManagerFactory()
 
         public void OnConfigChanged(object? sender, HotReloadEventArgs args)
         {
+            //First use of GetConfig during hot reload, in order to do validation of
+            //config file before any changes are made for hot reload.
+            //In case validation fails, an exception will be thrown and hot reload will be canceled.
+            _runtimeConfigProvider.GetConfig();
```

Comment:
Is this really the first use of GetConfig? why is this inside an OnConfigChanged of one particular class? Can we move it before OnConfigChanged function of this class is called?

Reply:
It is the first use after the creating the new config, I put it in that class specifically because it is the first one to be loaded and in order to not allow the program to start saving new information that is still not validated.
OnConfigChanged is called from `RuntimeConfigLoader` which calls the previous function from `FileSystemRuntimeConfigLoader` and neither of them has an object `RuntimeConfigProvider` that would allow us to call GetConfig.

Reply:
This code shows the event handler event raising and then lastly has the change token notification:
https://github.com/Azure/data-api-builder/blob/c9aa87b0257507b1ea223953cfc8b1c76928610b/src/Config/RuntimeConfigLoader.cs#L81-L91

In the last sentence of last response, What I wasn't sure of was whether `RaiseChanged(); ` for change token notification was a blocking call. Aaron explained that the event handler event raising was blocking. That's an important distinction because you'd want to move `RaiseChanged()` to come before ` OnConfigChangedEvent()` but you'd also want the operations trigger by `RaiseChanged()` (Config Validation that you add in this PR for hot reload) to complete before ` OnConfigChangedEvent()` code is executed

Reply:
From my testing it looks like RaiseChanged completes its execution before OnConfigChangedEvent starts